contracts/sol6/wrappers/KyberRateHelper.sol

```diff
@@ -207,4 +232,112 @@ contract KyberRateHelper is IKyberRateHelper, WithdrawableNoModifiers, Utils5 {
             );
         }
     }
+
+    function getSpreadInfo(IERC20 token, uint256 optionalAmountWei)
+        public
+        view
+        returns (bytes32[] memory reserves, uint256[] memory spreads)
+    {
+        (
+            bytes32[] memory buyReserves,
+            uint256[] memory buyRates,
+            bytes32[] memory sellReserves,
+            uint256[] memory sellRates
+        ) = getReservesRates(token, optionalAmountWei);
+
+        uint256[] memory revertReserveIndex = new uint256[](buyReserves.length);
+        uint256 validReserve = 0;
+        for (uint256 i = 0; i < buyRates.length; i++) {
+            if (buyRates[i] == 0) {
+                continue;
+            }
+
+            for (uint256 j = 0; j < sellRates.length; j++) {
+                if (sellReserves[j] == buyReserves[i]) {
+                    if (sellRates[j] == 0) break;
+                    revertReserveIndex[i] = j;
```

Comment:
I don't follow
when j != i?

Reply:
If we add token by listing 2 side at 1 transaction, i is always equals j
But since:
- we list token A for e2t, list token B for both sides and list token A for t2e. The order can be mess up. So it will better for flexibility, I think
- This is a read function, so add few extra gas will not matter

Reply:
Yes
Gas is not an issue. I Agree

I understand motivation now. I think it didn't so far.
but its a good idea. it could happen.

please write a short note about this in the code